scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,567 @@
+        CMAKE_FLAGS_BUILD_ALL,
+        [TEST_ALL]
+    )
+    tasks << build_and_test(
+        "debian-stable-minimal",
+        DOCKER_IMAGES.stretch_minimal,
+        [:],
+        [TEST_ALL]
+    )
+    return tasks
+}
+
+def build_todo() {
+    def test_name = "todo"
+    def open_task_patterns = '''\
+**/*.c, **/*.h, **/*.hpp, **/*.cpp,\
+**/CMakeLists.txt, **/Dockerfile*, Jenkinsfile*
+'''
+    return withDockerEnv(test_name, DOCKER_IMAGES.stretch_doc) {
+        sh "sloccount --duplicates --wide --details ${WORKSPACE} > sloccount.sc"
+        step([$class: 'SloccountPublisher', ignoreBuildFailure: true])
+        openTasks pattern: open_task_patterns,
+                  high: 'XXX',
+                  normal: 'FIXME',
+                  low: 'TODO'
+        archive(["sloccount.sc"])
+        deleteDir()
+    }
+}
+
+def build_doc() {
+    def test_name = "doc"
+    cmake_flags = [
+        'BUILD_PDF': 'ON',
+        'BUILD_FULL': 'OFF',
+        'BUILD_SHARED': 'OFF',
+        'BUILD_STATIC': 'OFF',
+        'BUILD_TESTING': 'OFF'
+    ]
+    return withDockerEnv(test_name, DOCKER_IMAGES.stretch_doc) {
+        dir('build') {
+            deleteDir()
+            cmake(env.WORKSPACE, cmake_flags)
+            sh "make html latex man pdf"
+        }
+        warnings parserConfigurations: [
+            [parserName: 'Doxygen', pattern: 'build/doc/doxygen.log']
+        ]
+        // TODO don't write to latest on PR's
+        sshPublisher(publishers: [
+          sshPublisherDesc(
+            configName: 'doc.libelektra.org',
+            transfers: [
+              sshTransfer(
+                sourceFiles: 'build/doc/latex/*',
+                removePrefix: 'build/doc/',
+                remoteDirectory: 'api/latest'
+              ),
+              sshTransfer(
+                sourceFiles: 'build/doc/man/*',
+                removePrefix: 'build/doc/',
+                remoteDirectory: 'api/latest'
+              ),
+              sshTransfer(
+                sourceFiles: 'doc/help/*.html',
+                removePrefix: 'doc/help/',
+                remoteDirectory: 'help'
+              )
+          ])
+        ])
+        deleteDir()
+    }
+}
+
+/* Helper to generate an asan enabled test */
+def build_and_test_asan(test_name, image, extra_cmake_flags = [:]) {
+    def cmake_flags = CMAKE_FLAGS_BASE +
+                      CMAKE_FLAGS_ASAN +
+                      extra_cmake_flags
+    return withDockerEnv(test_name, image) {
+        dir('build') {
+            deleteDir()
+            cmake(env.WORKSPACE, cmake_flags)
+            sh "make"
+            def llvm_symbolizer = sh(returnStdout: true,
+                                     script: 'which llvm-symbolizer').trim()
+            withEnv(["ASAN_OPTIONS='symbolize=1'",
+                     "ASAN_SYMBOLIZER_PATH=${llvm_symbolizer}"]){
+                ctest()
+            }
+        }
+    }
+}
+
+/* Helper to generate a typical elektra test environment
```

Comment:
This documentation is now useful, please also add similar documentation to all other vars and functions.

I think it would be useful if you use doxygen style (in particular @param) even if doxygen cannot parse it. But we are used in reading function docu in that way.

For the bigger picture (how jenkinsfiles and Dockerfiles interact and similar) a separate README.md would be useful. The current intro is not enough to understand how to read this file. The README.md can be done in a later PR.